### PR TITLE
adding NotificationHub binding

### PR DESCRIPTION
**Pull request by pragnagopa**
adding NotificationHub out binding

### CODE REVIEW

**Comment by pragnagopa** [comment]
@soninaren , @alrod can you review this ignoring the white spaces: https://github.com/Azure/azure-webjobs-sdk-templates/pull/57/files?w=0 as i formatted the file

File: Bindings/bindings.json [hunk 1]
```diff
@@ -6,473 +6,488 @@
...
+          "defaultValue": "{id}",
+          "required": true,
+          "label": "What is the Id for the record?",
+          "help": "Please enter the id for the record to retrieve."
+        }
+      ]
+    },
+    {
+      "type": "notificationHub",
+      "displayName": "Azure NotificationHub",
```
Comment (soninaren) [inline Bindings/bindings.json]: Nit: Azure Notification Hub

File: Bindings/bindings.json [hunk 2]
```diff
@@ -6,473 +6,488 @@
...
-                },
-                {
-                    "name": "runOnStartup",
-                    "value": "boolean",
-                    "defaultValue": false,
-                    "label": "Should this function run on start up?",
-                    "help": "When true, your function will run on startup. This may be useful for debugging."
-                }
-            ]
+  "bindings": [
```
Comment (mathewc) [inline Bindings/bindings.json]: What have we decided on spacing normalization? My local settings are also different that what is checked into this repo

Comment (soninaren) [inline Bindings/bindings.json]: It was 4 spaces

File: Bindings/bindings.json [hunk 3]
```diff
@@ -6,473 +6,488 @@
...
+      "direction": "out",      
+      "settings": [
+        {
+          "name": "tagExpression",
+          "value": "string",
+          "defaultValue": "tag1||tag2",
+          "required": false,
+          "label": "Optional.Tag expression for sending push notification",
+          "help": "Tag expressions enable you to target specific sets of devices/registrations, when sending a push notification through Notification Hubs."
         }
```
Comment (soninaren) [inline Bindings/bindings.json]: we need to have a setting like this https://github.com/pragnagopa/azure-webjobs-sdk-templates/blob/nhOut/Bindings/bindings.json#L426-L431
to select Notification hub

File: Bindings/bindings.json [hunk 4]
```diff
@@ -6,473 +6,504 @@
...
-                    "required": true,
-                    "label": "What is the Id for the record?",
-                    "help": "Please enter the id for the record to retrieve."
-                }
-            ]
+          "name": "connection",
+          "value": "string",
+          "resource": "NotificationHub",
+          "required": true,
+          "label": "Which NotificationHub namespace should be used?",
```
Comment (soninaren) [inline Bindings/bindings.json]: Is this Nh namespace or connection string?

Comment (pragnagopa) [inline Bindings/bindings.json]: setting is connection. The value of the connection string is for the NH namespace

Comment (soninaren) [inline Bindings/bindings.json]: May be something like "Connection string for the NH namespace" might be more accurate.

**Comment by soninaren** [comment]
Looks good merging it in

**Pull request merged**